src/sudoku/singlarobin/react/component/Generate/index.jsx

```diff
@@ -0,0 +1,193 @@
+        }
+    }, [started]);
+
+    const handleFormSubmit = () => setStarted(!started);
+
+    const handleUndo = () => {
+        if (undo.length === 0) {
+            alert('Not Possible');
+            return;
+        }
+        let operation = undo[undo.length - 1];
+
+        let index = operation['index'];
+        let currentGame = game;
+        currentGame[index] = operation['prev'];
+
+        let updatedUndoList = undo.slice(0, -1);
+        let updatedRedoList = [...redo, operation];
+
+        let sudoku = JSON.parse(localStorage.getItem('sudoku'));
+        sudoku[1]['game'] = currentGame;
+        sudoku[2]['undo'] = updatedUndoList;
+        sudoku[3]['redo'] = updatedRedoList;
+        localStorage.setItem('sudoku', JSON.stringify(sudoku));
+
+        setUpdatedIndex(index);
+        setUndo(updatedUndoList);
+        setRedo(updatedRedoList);
+        setGame(currentGame);
+    };
+
+    const handleRedo = () => {
+        if (redo.length === 0) {
+            alert('Not Possible');
+            return;
+        }
+        let operation = redo[redo.length - 1];
+        let index = operation['index'];
+        let currentGame = game;
+        currentGame[index] = operation['current'];
+
+        let updatedRedoList = redo.slice(0, -1);
+        let updatedUndoList = [...undo, operation];
+        let sudoku = JSON.parse(localStorage.getItem('sudoku'));
+        sudoku[1]['game'] = currentGame;
+        sudoku[2]['undo'] = updatedUndoList;
+        sudoku[3]['redo'] = updatedRedoList;
+        localStorage.setItem('sudoku', JSON.stringify(sudoku));
+
+        setUpdatedIndex(index);
+        setUndo(updatedUndoList);
+        setRedo(updatedRedoList);
+        setGame(currentGame);
+    };
+    const handleReset = () => {
+        localStorage.setItem('sudoku', JSON.stringify([]));
+        setStarted(false);
+    };
+
+    const generateSudokuApiCall = () => {
+        setLoading(true);
+        fetch(
+            SUDOKU_API + `diff=${DIFFICULTY[difficulty]}&stype=list&solu=true`,
+            {
+                method: 'GET',
+                headers: {
+                    'x-rapidapi-host': 'sudoku-board.p.rapidapi.com',
+                    'x-rapidapi-key': process.env.REACT_APP_SUDOKU_API_KEY,
+                },
+            }
+        )
+            .then((response) => response.json())
+            .then((data) => {
+                let sudoko = [];
+                sudoko.push(
+                    ...[
+                        { solution: data.response['solution'].flat() },
+                        { game: data.response['unsolved-sudoku'].flat() },
+                        { undo: [] },
+                        { redo: [] },
+                    ]
+                );
+                localStorage.setItem('sudoku', JSON.stringify(sudoko));
+
+                setGame(data.response['unsolved-sudoku'].flat());
+                setSolution(data.response['solution'].flat());
+                setLoading(false);
```

Comment:
setLoading can be taken out of try catch block

Reply:
added in finally block